projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm.e2e-spec.ts

```diff
@@ -0,0 +1,239 @@
+import * as addressBook from '../../helpers/address-book';
+import { login } from '../../helpers/auth-forms';
+import * as checkout from '../../helpers/checkout-flow';
+import { fillShippingAddress } from '../../helpers/checkout-forms';
+import * as consent from '../../helpers/consent-management';
+import { getErrorAlert } from '../../helpers/global-message';
+import * as loginHelper from '../../helpers/login';
+import * as profile from '../../helpers/update-profile';
+import { getSampleUser } from '../../sample-data/checkout-flow';
+import { clearAllStorage } from '../../support/utils/clear-all-storage';
+import {
+  interceptDelete,
+  interceptGet,
+  interceptPost,
+} from '../../support/utils/intercept';
+
+let customer: any;
+
+context('Assisted Service Module', () => {
+  before(() => {
+    clearAllStorage();
+
+    cy.visit('/');
+
+    customer = getSampleUser();
+    checkout.registerUser(false, customer);
+  });
+
+  describe('Customer Support Agent - Emulation', () => {
+    it('should test customer emulation', () => {
+      // storefront should have ASM UI disabled by default
+      checkout.visitHomePage();
+      cy.get('cx-asm-main-ui').should('not.exist');
+
+      cy.log('--> Agent login');
+      checkout.visitHomePage('asm=true');
+      cy.get('cx-asm-main-ui').should('exist');
+      cy.get('cx-asm-main-ui').should('be.visible');
+
+      agentLogin();
+
+      startCustomerEmulation();
+
+      cy.log('--> Update personal details');
+      cy.selectUserMenuOption({
+        option: 'Personal Details',
+      });
```

Comment:
In the past, I have seen `cy.selectUserMenuOption` being the cause of some flakiness.
If ever we have doubts that it could be the case, we could use `cy.visit` instead since the goal of this test is not to test the menu navigation

Reply:
Good one. Done.